### PR TITLE
WEB-1701 Unable to apply Vouchers in Staging

### PR DESCRIPTION
**Pull request by lukyfebrianto**
**Description:** Unable to apply Vouchers in Staging

**JIRA:**  https://liv-it.atlassian.net/browse/WEB-1701

**Merge deadline:** N/A

**Testing instructions:**

Steps to reproduce:
1. Create License (mine:Mk9qXoero4hWeq1MvdRWxwYexMVs) 
2. Apply new license in a new course (Teacher: elvana+119@labster.com) 
3. Create vouchers 
4. Login as student (Student: elvana+50@labster.com) 
5. Enter Voucher (Voucher:GF8JEGN2W2 or 8M4GFCBOWI)

**Reviewers:**
- [x] @TamoshaytisV  
- [x] @leha2000 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: lms/djangoapps/labster_vouchers/views.py [hunk 1]
```diff
@@ -118,6 +121,15 @@ def activate_voucher(request):
 
     course_license = course_licenses[0]
     course_id = course_license.course_id
+
+    # check invitation_only on course_overview is updated
+    course_key = SlashSeparatedCourseKey.from_deprecated_string(unicode(course_id))
```
Comment (TamoshaytisV) [inline lms/djangoapps/labster_vouchers/views.py]: can we assign `unicode(course_id)` to variable only once and use the result?

**Review by TamoshaytisV — CHANGES_REQUESTED** [review]
Luky, i think we have this situation because i've set "Invitation only" on staging. Even if we set that flag to `false` we could still have it cached. But on prod we have no such situation. So my question is: isn't it better to have some kind of management command or execute code in shell instead of updating cache every voucher apply? What do you think?

File: lms/djangoapps/labster_vouchers/views.py (hunk 1 above)
Comment (lukyfebrianto) [reply lms/djangoapps/labster_vouchers/views.py]: Ahh, I see so it isn't common situation. Yes agree with it but isn't better to update `invitation_only` through admin panel  i.e: https://staging-studio-courses.labster.com/admin/course_overviews/courseoverview/ ?

**Review by leha2000 — APPROVED** [review]

**Comment by lukyfebrianto** [comment]
FYI: 
I do some test that @TamoshaytisV told me as follows,
- to turn it on again for course in studio
- confirm that vouchers could not be applied (failed)
- turn off flag in studio 
- turn off flag in course overview 
- confirm vouchers work 

I found that step "confirm that vouchers could not be applied" not working so I change the code on studio so when user update through settings on studio it will synchronize data between course module and course_overview so data course_overview will always up to date.

File: cms/djangoapps/contentstore/views/course.py [hunk 2]
```diff
@@ -1207,6 +1208,16 @@ def advanced_settings_handler(request, course_key_string):
                         # now update mongo
                         modulestore().update_item(course_module, request.user.id)
 
+                        # Throw away old versions of CourseOverview and CourseOverviewTab - added by labster
+                        # as they might contain stale data then create new one.
+                        try:
+                            course_overview = CourseOverview.objects.get(id=course_key)
```
Comment (lukyfebrianto) [inline cms/djangoapps/contentstore/views/course.py:1214]: Check if course data is exist in courseOverview model

**Pull request closed without merging**